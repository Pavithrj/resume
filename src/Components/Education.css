.educationcontainer {
    margin-left: 10px;
}

/* .educationstyle {
    display: flex;
} */

.educationdetails{display: flex;}

.educationtimeperiod {
    width: 110px;
}

.stream {
    font-weight: bold;
}
.educationplacegrade{
    font-weight: lighter;

}