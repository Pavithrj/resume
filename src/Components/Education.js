import { useContext } from 'react';
import './Education.css';
import ProfileContext from './ProfileContext';

function Education() {
    const profileData = useContext(ProfileContext);
    const { education } = profileData;
    return (
        <div className='educationcontainer'>
            <h3 className="education">Education</h3>
            <div>
                <div className='educationstyle'>
                    {education.map(item => {
                        return (
                            <div className='educationdetails'>
                                <div className='educationtimeperiod'>
                                    {item.timePeriod}
                                </div>
                                <div className='stream'>
                                    {item.stream}
                                    <div className='educationplacegrade'>
                                        {item.place}
                                        <br />
                                        {item.grade}
                                    </div>
                                </div>
                            </div>
                        )
                    })}
                </div>
            </div>
        </div>
    )
}

export default Education;