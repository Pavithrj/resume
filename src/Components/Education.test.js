import { render } from "@testing-library/react";
import ProfileContext from './ProfileContext';
import profileData from '../Details.json';
import Education from "./Education";

describe('Contact Component', () => {
    const customRender = (ui, { providerProps, ...renderOptions }) => {
        return render(
            <ProfileContext.Provider value={profileData}>{ui}</ProfileContext.Provider>,
            renderOptions,
        )
    }
    it('should print contact details', () => {
        const { asFragment } = customRender(<Education />, {
            value: profileData
        });
        expect(asFragment()).toMatchSnapshot();
    })
})