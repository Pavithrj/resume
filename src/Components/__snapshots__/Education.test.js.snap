// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should print contact details 1`] = `
<DocumentFragment>
  <div
    class="educationcontainer"
  >
    <h3
      class="education"
    >
      Education
    </h3>
    <div>
      <div
        class="educationstyle"
      >
        <div
          class="educationdetails"
        >
          <div
            class="educationtimeperiod"
          >
            June 2010 - July 2013
          </div>
          <div
            class="stream"
          >
            B.E Electronics and Telecommunication
            <div
              class="educationplacegrade"
            >
              Amaravati University - Maharashtra
              <br />
              GPA: 72.44
            </div>
          </div>
        </div>
        <div
          class="educationdetails"
        >
          <div
            class="educationtimeperiod"
          >
            June 2007 - May 2010
          </div>
          <div
            class="stream"
          >
            Diploma Electronics and Telecommunication
            <div
              class="educationplacegrade"
            >
              Government Polytechnic - Amaravati, Maharashtra
              <br />
              GPA: 88.27
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
